### ISSUE TITLE
No feature to add custom metric (mAP) for model training

### ISSUE THREAD

**Issue by magda1428** [matterport/Mask_RCNN#752]
Hi , 

I would like to customize metrics by adding  mAP for training.
Usually I define custom function (for instance _calculate_mAP_ ) and then pass it to compile method as showed below:

```
model.compile(optimizer='rmsprop',
              loss='binary_crossentropy',
              metrics=['accuracy', calculate_mAP])
```
As far as I understand,  in model.py module there is no argument to pass to keras compile method (copied below from the source code)

```
   self.keras_model.compile(
            optimizer=optimizer,
            loss=[None] * len(self.keras_model.outputs))
```

I would be thankful if you could extend your repo with such an option.

Thank you in advance!

**Comment by vgthengane** [comment]
Hi folks,

Any updates on this?
